### PR TITLE
feat: implement mvi get item and get item metadata methods

### PR DESCRIPTION
**Pull request by malandis**
Implements MVI `GetItemBatchAsync` and `GetItemMetadataBatchAsync` methods, response types, and adds integration tests. Also extracts metadata dictionary equality and hash code to a central extension method.

### CODE REVIEW

**Review by bruuuuuuuce — DISMISSED** [review]
:shipit:

File: src/Momento.Sdk/IPreviewVectorIndexClient.cs [hunk 1]
```diff
@@ -132,6 +132,57 @@ public interface IPreviewVectorIndexClient : IDisposable
...
+    ///    return;
+    /// }
+    /// </code>
+    /// </returns>
+    public Task<GetItemBatchResponse> GetItemBatchAsync(string indexName, IEnumerable<string> ids);
+
+    /// <summary>
+    /// Gets metadata for a batch of items from a vector index by ID.
+    /// </summary>
+    /// <param name="indexName">The name of the vector index to get the items from.</param>
```
Comment (nand4011) [inline src/Momento.Sdk/IPreviewVectorIndexClient.cs:164]:
```suggestion
    /// <param name="indexName">The name of the vector index to get the item metadata from.</param>
```

File: src/Momento.Sdk/Responses/Vector/GetItemMetadataResponse.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Momento.Sdk.Exceptions;
+using Momento.Sdk.Messages.Vector;
+using Momento.Sdk.Internal.ExtensionMethods;
+
+namespace Momento.Sdk.Responses.Vector;
+
+/// <summary>
+/// Parent response type for a get item batch request. The
```
Comment (nand4011) [inline src/Momento.Sdk/Responses/Vector/GetItemMetadataResponse.cs:10]:
```suggestion
/// Parent response type for a get item metadata batch request. The
```

**Review by nand4011 — DISMISSED** [review]
A couple of doc nits, but LGTM.

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**